File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def calHeight(self , root):
        if not root :
            return 1
        left = self.calHeight(root.left)
        right = self.calHeight(root.right)
        return 1 + max(left,right) 
    def isBalanced(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: bool
        """
        if not root :
            return True
        q = [root]
        visited = set()
        while q :
            node = q.pop(0)
            left = self.calHeight(node.left)
            right = self.calHeight(node.right)
            if abs(right - left ) > 1 :
                return False
            visited.add(node)
            if node.left and node.left not in visited :
                q.append(node.left)
            if node.right and node.right not in visited :
                q.append(node.right)
        return True 

        

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int checkHeight(TreeNode node){
        if(node == null)
            return 1;
        int left = checkHeight(node.left);
        int right = checkHeight(node.right);
        if(Math.abs(left - right) > 1)
            return -1;
        if(left == -1 || right == -1)
            return -1;
        return 1 + Math.max(left , right);
    }

    public boolean isBalanced(TreeNode root) {
        int result = checkHeight(root);
        if(result == -1)
            return false;
        return true;
       
        
    }
}